x-pack/plugins/security_solution/public/common/components/markdown_editor/plugins/lens/plugin.tsx

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiComboBox,
+  EuiModalBody,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiMarkdownEditorUiPlugin,
+  EuiCodeBlock,
+  EuiSpacer,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiFormRow,
+  EuiDatePicker,
+  EuiDatePickerRange,
+} from '@elastic/eui';
+import React, { useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import moment, { Moment } from 'moment';
+
+import { TypedLensByValueInput } from '../../../../../../../lens/public';
+import { useKibana } from '../../../../lib/kibana';
+import { LensMarkDownRenderer } from './processor';
+import { ID } from './constants';
+import * as i18n from './translations';
+
+const ModalContainer = styled.div`
+  width: ${({ theme }) => theme.eui.euiBreakpoints.m};
+`;
+
+interface LensEditorProps {
+  id?: string | null;
+  title?: string | null;
+  startDate?: Moment | null;
+  endDate?: Moment | null;
+  onClosePopover: () => void;
+  onInsert: (markdown: string, config: { block: boolean }) => void;
+}
+
+const LensEditorComponent: React.FC<LensEditorProps> = ({
+  id,
+  title,
+  startDate: defaultStartDate,
+  endDate: defaultEndDate,
+  onClosePopover,
+  onInsert,
+}) => {
+  const soClient = useKibana().services.savedObjects.client;
+  const [lensOptions, setLensOptions] = useState<Array<{ label: string; value: string }>>([]);
+  const [selectedOptions, setSelectedOptions] = useState<Array<{ label: string; value: string }>>(
+    id && title ? [{ value: id, label: title }] : []
+  );
+  const [lensSavedObjectId, setLensSavedObjectId] = useState<string | null>(id ?? null);
+  const [startDate, setStartDate] = useState<Moment | null>(
+    defaultStartDate ? moment(defaultStartDate) : moment().subtract(7, 'd')
+  );
+  const [endDate, setEndDate] = useState<Moment | null>(
+    defaultEndDate ? moment(defaultEndDate) : moment()
+  );
+
+  useEffect(() => {
+    const fetchLensSavedObjects = async () => {
+      const { savedObjects } = await soClient.find<TypedLensByValueInput['attributes']>({
+        type: 'lens',
+        perPage: 1000,
```

Comment:
Do we gonna fetch 1000 docs per entry? What will happened if we have a case with 10 lens inside? Are we gonna fetch 10*1000 documents?